### ISSUE TITLE
container-validator: regex to validate link.id and other attributes wrong use of range

### ISSUE THREAD

**Issue by italovalcy** [atlanticwave-sdx/sdx-continuous-development#99]
Hi,

On the `container-validator`, the `validator.yml` includes some regex to validate `link.id`, name, etc, and some of them use the "hyphen" character wrongly. A hyphen is used to indicate a range of characters. It should be escaped or used at the end of the regex; otherwise, it has a different semantic (range or characters) from what was expected there (actual hyphen character).

Here are some changes proposed:

```
diff --git a/data-plane/container-validator/validator.yml b/data-plane/container-validator/validator.yml
index 9a05bb8..1325217 100755
--- a/data-plane/container-validator/validator.yml
+++ b/data-plane/container-validator/validator.yml
@@ -77,7 +77,7 @@ paths:
                       properties:
                         id:
                           type: string
-                          pattern: '^((urn:sdx:node:)[A-Za-z_.-\:]*$)'
+                          pattern: '^((urn:sdx:node:)[A-Za-z_.\:-]*$)'
                         name:
                           type: string
                           minLength: 3
@@ -107,12 +107,12 @@ paths:
                       properties:
                         id:
                           type: string
-                          pattern: '^((urn:sdx:link:)[A-Za-z_.-\:]*$)'
+                          pattern: '^((urn:sdx:link:)[A-Za-z_.\:-]*$)'
                         name:
                           type: string
                           minLength: 3
                           maxLength: 30
-                          pattern: '^[A-Za-z0-9_.-/]*$'
+                          pattern: '^[A-Za-z0-9_./-]*$'
                         ports:
                           type: array
                           minItems: 1
@@ -261,7 +261,7 @@ components:
       properties:
         id:
           type: string
-          pattern: '^((urn:sdx:port:)[A-Za-z_.-\:]*$)'
+          pattern: '^((urn:sdx:port:)[A-Za-z_.\:-]*$)'
         name:
           type: string
           minLength: 3
@@ -269,7 +269,7 @@ components:
           pattern: '^[A-Za-z0-9_.-]*$'
         node:
           type: string
-          pattern: '^((urn:sdx:node:)[A-Za-z_.-\:]*$)'
+          pattern: '^((urn:sdx:node:)[A-Za-z_.\:-]*$)'
         type:
           type: string
           enum: ['100FE','1GE','10GE','25GE','40GE','50GE','100GE','400GE','Other']
```

**Comment by italovalcy** [comment]
One side effect of the above proposed change is that `pattern: '^((urn:sdx:node:)[A-Za-z_.-\:]*$)'` was ended up allowing numbers on the regex, while moving hyffen to the end fix this side effect and creates the problem of now allowing numbers. Thus, the correct change would be:

```
diff --git a/data-plane/container-validator/validator.yml b/data-plane/container-validator/validator.yml
index 9a05bb8..fb296bb 100755
--- a/data-plane/container-validator/validator.yml
+++ b/data-plane/container-validator/validator.yml
@@ -49,12 +49,12 @@ paths:
               properties:
                   id:
                     type: string
-                    pattern: '^((urn:sdx:topology:)[A-Za-z_.:-]*$)'
+                    pattern: '^((urn:sdx:topology:)[A-Za-z0-9_.:-]*$)'
                   name:
                     type: string
                     minLength: 3
                     maxLength: 30
-                    pattern: '^[A-Za-z_.-]*$'
+                    pattern: '^[A-Za-z0-9_.-]*$'
                   version:
                     type: integer
                     format: int64
@@ -77,7 +77,7 @@ paths:
                       properties:
                         id:
                           type: string
-                          pattern: '^((urn:sdx:node:)[A-Za-z_.-\:]*$)'
+                          pattern: '^((urn:sdx:node:)[A-Za-z0-9_.\:-]*$)'
                         name:
                           type: string
                           minLength: 3
@@ -107,12 +107,12 @@ paths:
                       properties:
                         id:
                           type: string
-                          pattern: '^((urn:sdx:link:)[A-Za-z_.-\:]*$)'
+                          pattern: '^((urn:sdx:link:)[A-Za-z0-9_.\:-]*$)'
                         name:
                           type: string
                           minLength: 3
                           maxLength: 30
-                          pattern: '^[A-Za-z0-9_.-/]*$'
+                          pattern: '^[A-Za-z0-9_./-]*$'
                         ports:
                           type: array
                           minItems: 1
@@ -261,7 +261,7 @@ components:
       properties:
         id:
           type: string
-          pattern: '^((urn:sdx:port:)[A-Za-z_.-\:]*$)'
+          pattern: '^((urn:sdx:port:)[A-Za-z0-9_.\:-]*$)'
         name:
           type: string
           minLength: 3
@@ -269,7 +269,7 @@ components:
           pattern: '^[A-Za-z0-9_.-]*$'
         node:
           type: string
-          pattern: '^((urn:sdx:node:)[A-Za-z_.-\:]*$)'
+          pattern: '^((urn:sdx:node:)[A-Za-z0-9_.\:-]*$)'
         type:
           type: string
           enum: ['100FE','1GE','10GE','25GE','40GE','50GE','100GE','400GE','Other']
```

**Comment by italovalcy** [comment]
This issue was fixed by PR #103
